feat(cli): introduce default entrypoints (#4257)
winglang/wing: docs/docs/06-tools/01-cli.md

npm i -g winglang
```

Usage:

```sh
$ wing [command] [options]
```

## Run: `wing run|it`

You can use the `run` command (or `it`) when you want to interact with your Wing program in the
[Wing Console](/docs/start-here/local).

Usage:

```sh
$ wing run|it [entrypoint]
```

`[entrypoint]` is a valid entrypoint for a Wing program. An entrypoint can be either source code or compiled. If it's source code, the entrypoint refers to a file named `main.w` or ending with `.main.w`/`.test.w`, whereas if it's compiled the entrypoint refers to a directory whose name ends with `.wsim`.

:::note Default Entrypoint

> It's possible to execute `wing run|it` without specifying any entrypoint, in which case the CLI looks for exactly one file named `main.w` or ending with `.main.w` in the current directory. If no files or more than one file are found, the CLI throws an error.

Reviewer: This can be shortened, no need to mention the error IMO

```suggestion
By default, `wing run|it` will look for exactly one file named `main.w` or ending with `.main.w`
```

:::

## Compile: `wing compile`

You can use the `compile` command to compile a Wing program into a deployable artifact.

```sh
$ wing compile [entrypoint] --target <target>
```

`[entrypoint]` specifies the entrypoint file to compile. A file is considered a valid entrypoint if its name is `main.w` or if it ends with `.main.w`/`.test.w`.

:::note Default Entrypoint

> It's possible to execute `wing compile` without specifying any entrypoint, in which case the CLI looks for exactly one file named `main.w` or ending with `.main.w` in the current directory. If no files or more than one file are found, the CLI throws an error.

Reviewer: Same as above

```suggestion
By default, `wing compile` will look for exactly one file named `main.w` or ending with `.main.w`
```

:::

> The `--target` option is **required**, and specifies the target platform to compile for. The
> following targets are supported:

Reviewer: ```suggestion
The --target option specifies the target platform to compile for. The default target is `sim`.
The following targets are supported:
```

* `tf-aws` - Terraform/AWS
* `tf-azure` - Terraform/Azure

> :::note Default Target
> 
> It's possible to execute `wing compile` without specifying any target, in which case the CLI defaults to `sim` target.
> 
> :::
> 

Reviewer: No need for this if above suggestion is accepted

```suggestion
```

```sh
$ export CDK_STACK_NAME="my-project"

> $ wing compile [entrypoint] --target awscdk

Reviewer: Why reorder?

Author: It looked better to me to follow the same order everywhere with the `[entrypoint]` soon after the command (`wing run|it` doesn't have an option for example). If you think that it was better the other way around, I can revert this change.

Reviewer: I don't feel strongly about it

Usage:

```sh
$ wing test [entrypoint...]
```

`[entrypoint...]` specifies the entrypoint list of files that will be compiled and tested. Valid entrypoint names are `main.w`, `<file>.main.w` and `<file>.test.w`.

For example ([test_bucket.test.w](https://github.com/winglang/wing/tree/main/examples/tests/valid/test_bucket.test.w)):

> :::default entrypoint...

Reviewer: ```suggestion
:::note Default Entrypoint(s)
```

> It's possible to execute `wing test` without specifying any entrypoint, in which case the CLI looks for all files ending with `.test.w` in the current directory.

Reviewer: ```suggestion
Executing `wing test` without specifying any entrypoint looks for all entrypoint files in the current directory.
```

Now, if we run the following command:

```sh
$ wing test test_bucket.test.w

> pass | test_bucket.testw | root/test:get

Reviewer: ```suggestion
pass | test_bucket.test.w | root/test:get
```
